google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorkerPool.java

```diff
@@ -234,47 +233,52 @@ public ApiFuture<AppendRowsResponse> append(StreamWriter streamWriter, ProtoRows
   public ApiFuture<AppendRowsResponse> append(
       StreamWriter streamWriter, ProtoRows rows, long offset) {
     // We are in multiplexing mode after entering the following logic.
-    ConnectionWorker connectionWorker =
-        streamWriterToConnection.compute(
-            streamWriter,
-            (key, existingStream) -> {
-              // Though compute on concurrent map is atomic, we still do explicit locking as we
-              // may have concurrent close(...) triggered.
-              lock.lock();
-              try {
-                // Stick to the existing stream if it's not overwhelmed.
-                if (existingStream != null && !existingStream.getLoad().isOverwhelmed()) {
-                  return existingStream;
-                }
-                // Try to create or find another existing stream to reuse.
-                ConnectionWorker createdOrExistingConnection = null;
-                try {
-                  createdOrExistingConnection =
-                      createOrReuseConnectionWorker(streamWriter, existingStream);
-                } catch (IOException e) {
-                  throw new IllegalStateException(e);
-                }
-                // Update connection to write stream relationship.
-                connectionToWriteStream.computeIfAbsent(
-                    createdOrExistingConnection, (ConnectionWorker k) -> new HashSet<>());
-                connectionToWriteStream.get(createdOrExistingConnection).add(streamWriter);
-                return createdOrExistingConnection;
-              } finally {
-                lock.unlock();
-              }
-            });
+    ConnectionWorker currentConnection;
```

Comment:
In general, I like this idea. We can reuse the connection more for the same StreamWriter.